### DISCUSSION TITLE
OpenAI text import

### DISCUSSION THREAD

**Discussion by stusemple** [mealie-recipes/mealie#4532] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

My girlfriend loves using instagram to find new recipes. At the moment however there is no easy way to import recipes from instagram. I can use an ‘instagram downloader’ iOS shortcut to obtain the free text from the post and I wonder if mealie could have a ‘text import’ option where it sends the free text to OpenAI, just like it does with an image, and OpenAI returns it in an API friendly format.

### Please provide a concise description of the feature that would resolve your issue.

Free text recipe import that uses OpenAI 

### Please consider and list out some caveats or tradeoffs made in your design decision

Could use existing OpenAI integration with a few tweaks to prompts. Cannot see any other downsides

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by JoTec2002** [comment]
Have a look at the discussions https://github.com/mealie-recipes/mealie/discussions/3846 and https://github.com/mealie-recipes/mealie/discussions/1474.

I mentioned in this discussions that I wrote a little docker container, that does exactly what you want (directly from the instagram link, including adding the instagram video as an asset to Mealie for Archiving) https://github.com/JoTec2002/InstagramToMealie.


If you don’t wanna use the extra container I found, that if you just post the instagram description into Mealie HTML Importer it works pretty well.

**Reply by stusemple** [reply]
I just got this working. Thanks so much for your help. What a great feature.
